tripy/examples/resent50/resnet50.ipynb

```diff
@@ -0,0 +1,529 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ResNet50 Image Classificiation with `tripy` - An Implementation and Demo\n",
+    "\n",
+    "This notebook aims to demonstrate the implementation of a ResNet50 model using `tripy`. We'll explore the architecture, \n",
+    "load pretrained weights, and run predictions on a sample dataset, showcasing how `tripy` can be used effectively for inference of ResNet50 based image classification model.\n",
+    "\n",
+    "### Objectives:\n",
+    "1. Implement ResNet50 using `tripy` and load pretrained weights.\n",
+    "2. Run predictions on sample images and visualize results.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Setup\n",
+    "\n",
+    "install neccessary libraries, load pretrained resnet50 weights and the inference dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install datasets matplotlib pillow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load necessary libraries\n",
+    "import torch\n",
+    "from transformers import ResNetForImageClassification, AutoImageProcessor\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "# Load the pretrained ResNet50 model from Hugging Face\n",
+    "resnet_pretrained = ResNetForImageClassification.from_pretrained(\"microsoft/resnet-50\")\n",
+    "resnet_pretrained.eval()\n",
+    "resnet_pretrained = resnet_pretrained.to('cuda')\n",
+    "\n",
+    "# Load the image processor\n",
+    "processor = AutoImageProcessor.from_pretrained(\"microsoft/resnet-50\")\n",
+    "\n",
+    "# Load a sample image from the dataset\n",
+    "dataset = load_dataset(\"comet-team/coco-500\", split=\"train\")\n",
+    "idx = 103\n",
+    "image = dataset[idx]['Image']\n",
+    "image"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
I don't use notebooks a lot - is there some convention on how code is split between cells? Should each conceptually different thing, e.g. loading weights, running inference, be in its own cell?

Reply:
Yeah, it is better if we seperate them so each cell is concise. Unless code is highly correlated like running forward path on image and showing the plot are fine to put on the same cell. But I think for model we should have Modules separate from weight loading and inference

Reply:
I reduced number of cells now only 3 cells
- 1 cell for defining the core convolution + Norm layers
- 1 for building blocks
- 1 for actual backbone and classifier wrapper